x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -165,7 +166,20 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
   // reset form when rule type changes
   useEffect(() => {
     reset({ resetValues: false });
-  }, [reset, ruleType]);
+    if (getFields().queryBar != null) {
```

Comment:
I think you can use `queryBar` from line 140? Don't think you have to re-pull it here.

Reply:
EDIT: Nvm, I see reset was called beforehand.